internal/codegen/cmd/defaultsconfig/v1.go

```diff
@@ -0,0 +1,321 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	codegen "github.com/aws/aws-sdk-go-v2/internal/codegen/defaults"
+	"log"
+	"math/big"
+	"sort"
+	"strings"
+	"text/template"
+)
+
+func renderPtrTimeMillisecond(context *generationContext, base interface{}, modifier Modifier) (string, error) {
+	context.AddImport("time", "")
+
+	value, err := applyBaseModifier(base, modifier)
+	if err != nil {
+		return "", err
+	}
+
+	i, ok := value.(int64)
+	if !ok {
+		return "", fmt.Errorf("expect int64, got %T", base)
+	}
+
+	return fmt.Sprintf("duration(%d*time.Millisecond)", i), nil
+}
+
+type ConfigurationValue struct {
+	FieldName string
+	Setter    func(context *generationContext, base interface{}, modifier Modifier) (string, error)
+}
+
+var supportedConfigKeys = map[string]ConfigurationValue{
+	"connectTimeoutInMillis": {
+		FieldName: "ConnectTimeout",
+		Setter:    renderPtrTimeMillisecond,
+	},
+	"tlsNegotiationTimeoutInMillis": {
+		FieldName: "TLSNegotiationTimeout",
+		Setter:    renderPtrTimeMillisecond,
+	},
+}
+
+// ModifierMap is a map of configuration keys to their respective modifier descriptor.
+type ModifierMap map[string]Modifier
+
+// Modifier is a union type of either a multiply, add, or override modification descriptor.
+type Modifier struct {
+	Multiply *json.Number
+	Add      *int64
+	Override interface{}
```

Comment:
Should all of these numbers be `json.Number` instead of `interface` so they are always known type and type casting isn't needed?  Could `Override` ever be a non-number type, e.g. bool?

Reply:
According the current specification `Add` will be an int64, but I can move this to a json.Number and enforce it via a conversion to ensure it wasn't overflowed or something.

`Override` is a special case since it can be applied be a string or numerical value depending on the key the modifier is targeting.

Reply:
Upon further inspection we are going for `add` I was already doing what I had stated.